### ISSUE TITLE
consistent-return don't recognize switch default case return as return

### ISSUE THREAD

**Issue by tinovyatkin** [eslint/eslint#8590]
false positive:

```js
function tes(a, b) {
  if (a) return 'Hello!';
  switch (b) {
     case 1:
         return 'tada';
    default:
         return 'boo'; 
   }
 }
```

**Comment by eslintbot** [comment]
Hi @tinovyatkin, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by not-an-aardvark** [comment]
I can't reproduce this issue, the rule does not report an error for that code on the latest version of ESLint.

**Comment by tinovyatkin** [comment]
Well the whole original code is:
```js
export function adjustHoursFromMeridiems(hours, meridiemStr) {
  if (typeof meridiemStr !== 'string') return hours;
  switch (MERIDIEMS.indexOf(meridiemStr)) {
    case 1: // PM
      if (hours >= 1 && hours < 12) return hours + 12;
      console.error('Invalid hours value %d', hours);
      return 0;

    case 0: // AM
      if (hours > 11 || hours < 1) return 0;
      break;

    default:
      console.assert('Invalid meridiem value! ', meridiemStr);
      return hours;
  }
}
```

**Comment by not-an-aardvark** [comment]
That looks like it's working as intended. If you reach the `break` statement in `case 0`, your function won't return a value.

**Comment by tinovyatkin** [comment]
Oops, sorry, my fault!
